sending-a-simple-message-to-a-service/README.md

```diff
@@ -0,0 +1,325 @@
+```
+
+The argument to the `new` expression is the URL for the backend service.
+
+5. Define a record corresponding to the payload from the backend service.
+
+```ballerina
+type Doctor record {|
+    string name;
+    string hospital;
+    string category;
+    string availability;
+    decimal fee;
+|};
+```
+
+The payload will be an array of JSON objects, where the structure of each JSON object matches this record. Note that you can use the "Paste JSON as record" VSCode command to generate the record if you have the JSON payload.
+
+6. Define the [HTTP service (REST API)](https://ballerina.io/learn/by-example/#rest-service) that has the resource that accepts user requests, retrieves relevant details from the backend service, and responds to the request. Use `/healthcare` as the service path (or the context) of the service which is attached to the listener listening on port `port`. Define an HTTP resource that allows the `GET` operation on resource path `/querydoctor` and accepts the `category` (corresponding to the specialization) as a path parameter.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() {
+        
+    }
+}
+```
+
+7. Implement the logic to retrieve and respond with relevant details.
+
+```ballerina
+service /healthcare on new http:Listener(port) {
+    resource function get querydoctor/[string category]() 
+            returns Doctor[]|http:NotFound|http:InternalServerError {
+        log:printInfo("Retrieving information", specialization = category);
+        
+        Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+        if resp is Doctor[] {
+            return resp;
+        }
+
+        if resp is http:ClientRequestError {
+            return <http:NotFound> {body: string `category not found: ${category}`};
+        }
+
+        return <http:InternalServerError> {body: resp.message()};
+    }
+}
+```
+
+- The `log:printInfo` statement [logs](https://ballerina.io/learn/by-example/#log) information about the request.
+
+```ballerina
+log:printInfo("Retrieving information", specialization = category);
+```
+
+- The call to the backend is done using a remote method call expression (using `->`) which distinguishes network calls from normal method calls. [Client data binding](https://ballerina.io/learn/by-example/http-client-data-binding/) is used to directly try and bind the JSON response on success to the expected array of records.
+
+```ballerina
+Doctor[]|http:ClientError resp = queryDoctorEP->/[category];
+```
+
+- Use the `is` check to decide the response based on the response to the client call. If the client call was successful and the respond payload was an array of `Doctor`s (as expected) directly return the array from the resource. If the request failed, send a "NotFound" response if the client call failed with a 4xx status code or return an "InternalServerError" response for other failures.
```

Comment:
```suggestion
- Use the `is` check to decide the response based on the response to the client call. If the client call was successful and the respond payload was an array of `Doctor`s (as expected), then, directly return the array from the resource. If the request failed, send a "NotFound" response if the client call failed with a 4xx status code or return an "InternalServerError" response for other failures.
```
We can add these messages also within backticks to differentiate them from the normal text.